RankingTests/RankingTests.cs

```diff
@@ -33,30 +36,67 @@ public void TestGetScore()
             bool actual = rank.Scores.Contains(score2);
             Assert.AreEqual(true, actual, "Test ranking can add score");
         }
+        //test: check if file can be loaded from
+        //expected: file is missing so cannot load (false)
         [Test]
-        public void TestLoadFileFalse()
+        public void TestCheckFileMissing()
         {
             Ranking rank = new Ranking();
-            bool actual = rank.LoadFile("scorenope.txt");
-            Assert.AreEqual(false, actual, "Test ranking can load score from file");
+            bool actual = rank.CheckFile("scorenope.txt");
+
+            Assert.AreEqual(false, actual, "Test ranking can detect missing file");
+        }
+        //test: check if file can be loaded from
+        //expected: file has more than 11 lines so cannot load (false)
+        [Test]
+        public void TestCheckFileValid1()
+        {
+            Ranking rank = new Ranking();
+            bool actual = rank.CheckFile("scoreover11.txt");
+
+            Assert.AreEqual(false, actual, "Test ranking can check for file validity");
         }
+        //test: check if file can be loaded from
+        //expected: file has non-digit chars so cannot load (false)
         [Test]
-        public void TestLoadFileTrue()
+        public void TestCheckFileValid2()
         {
             Ranking rank = new Ranking();
+            bool actual = rank.CheckFile("scorenondigit.txt");
+
+            Assert.AreEqual(false, actual, "Test ranking can check for file validity");
+        }
+        //test: check if ranking obj can read file content, line by line
+        //expected: concatenated string of first 3 lines of file 
+        [Test]
+        public void TestGetContentFromFile()
+        {
+            Ranking rank = new Ranking();
+            rank.LoadFile("scores.txt");
             rank.LoadFile("scores.txt");
-            bool actual = (rank.Scores.Length == 2);
-            Assert.AreEqual(true, actual, "Test ranking can load score from file");
+            string line = rank.GetLine();
+            string actual = "";
+            while (line != null)
+            {
+                actual += line;
+                line = rank.GetLine();
+            }
+            Assert.AreEqual("22015", actual, "Test ranking can print file content");
         }
+        //test: check if file is saved correctly
+        //expected: number of scores in the ranking obj 
         [Test]
         public void TestSaveFile1()
```

Comment:
I don't know how to read & write to file using something else.

Reply:
Let me try brb

Reply:
I'm not supposed not to use StreamReader, but we use it in a method and then test that method rather than to include StreamReader or StreamWriter in the test case.

Keep in mind that we should start with the empty method in TDD.

Reply:
Okay